File: src/ios/OpenAppLink.m
#import "OpenAppLink.h"
#import <Cordova/CDVPlugin.h>

@implementation OpenAppLink

- (void)open:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSString* url = [command.arguments objectAtIndex:0];

    if (url != nil && [url length] > 0) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:url];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end

File: src/ios swift/OpenAppLink.swift
//For ios, you need to set a swift version https://stackoverflow.com/a/52950615/6276471

@objc(OpenAppLink) class OpenAppLink : CDVPlugin {
    @objc(open:)
    func open(command: CDVInvokedUrlCommand) {
        let url_string = command.arguments[0] as? String ?? ""
        let url = URL(string: url_string)!

        if #available(iOS 10.0, *) {
            UIApplication.shared.open(url, options: [UIApplication.OpenExternalURLOptionsKey.universalLinksOnly : true]) { (success) in
                if (success) {
                    pluginResponse(message_string: "true")
                } else {
                    pluginResponse(message_string: "false")
                }
            }
        } else {
            pluginResponse(message_string: "false")
        }

        func pluginResponse(message_string: String) {
            let pluginResult = CDVPluginResult(
                status: CDVCommandStatus_OK,
                messageAs: message_string
            )

            self.commandDelegate!.send(
                pluginResult,
                callbackId: command.callbackId
            )
        }

    }
}
